dlpy/tests/datasources/sample_syntax_for_test.json

```diff
@@ -16,93 +16,98 @@
+    {"name": "R1B2", "layer": {"act": "relu", "type": "batchnorm"}, "sharedweights": null, "srclayers": ["R1C1"]},
+    {"name": "R1C2", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "sharedweights": null, "srclayers": ["R1B2"]},
+    {"name": "Res1", "layer": {"act": "identity", "type": "residual"}, "sharedweights": null, "srclayers": ["R1C2", "data"]}
   ],
   "resblock3" : [
-    {"name": "R1C1", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "srclayers": ["data"]},
-    {"name": "R1B1", "layer": {"act": "relu", "type": "batchnorm"}, "srclayers": ["R1C1"]},
-    {"name": "R1C2", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "srclayers": ["R1B1"]},
-    {"name": "R1B2", "layer": {"act": "relu", "type": "batchnorm"}, "srclayers": ["R1C2"]},
-    {"name": "Res1", "layer": {"act": "relu", "type": "residual"}, "srclayers": ["R1B2", "data"]}
+    {"name": "R1C1", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "sharedweights": null, "srclayers": ["data"]},
+    {"name": "R1B1", "layer": {"act": "relu", "type": "batchnorm"}, "sharedweights": null,"srclayers": ["R1C1"]},
+    {"name": "R1C2", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "sharedweights": null, "srclayers": ["R1B1"]},
+    {"name": "R1B2", "layer": {"act": "relu", "type": "batchnorm"}, "sharedweights": null, "srclayers": ["R1C2"]},
+    {"name": "Res1", "layer": {"act": "relu", "type": "residual"}, "sharedweights": null, "srclayers": ["R1B2", "data"]}
   ],
   "resblock4" : [
-    {"name": "R1C0", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": 1, "width": 1, "nfilters": 16, "type": "convo"}, "srclayers": ["data"]},
-    {"name": "R1B0", "layer": {"act": "identity", "type": "batchnorm"}, "srclayers": ["R1C0"]},
-    {"name": "R1C2", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "srclayers": ["data"]},
-    {"name": "R1B2", "layer": {"act": "relu", "type": "batchnorm"}, "srclayers": ["R1C2"]},
-    {"name": "R1C3", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": [3], "width": [3], "nfilters": 16, "type": "convo"}, "srclayers": ["R1B2"]},
-    {"name": "R1B3", "layer": {"act": "relu", "type": "batchnorm"}, "srclayers": ["R1C3"]},
-    {"name": "Res1", "layer": {"act": "relu", "type": "residual"}, "srclayers": ["R1B3", "R1B0"]}
+    {"name": "R1C0", "layer": {"includebias": false, "act": "identity", "stride": 1, "height": 1, "width": 1, "nfilters": 16, "type": "convo"}, "sharedweights": null, "srclayers": ["data"]},
```

Comment:
null? ok. I am not really happy about nulls here. I know you want to have something like this sharedweights:none, but I am wondering why we are explicitly enforcing this, what is the default value for sharedweights? let's discuss this